### PR TITLE
James/restore ls samples

### PR DESCRIPTION
**Pull request by JamesMBallard**
# Description

Add LocalServer category back to the sample viewer for Windows and Linux, if and only if local server is installed on the machine it's running on.

This logic was lost in the porting. There is no more `CppSampleManager` class, so I removed `virtual` to make it clearer there's no more overrides. 
https://github.com/Esri/arcgis-maps-sdk-samples-qt/blob/6d49d3fce7f8b89d72da7589aacd22c90d91f773/ArcGISRuntimeSDKQt_SampleViewers/ArcGISRuntimeSDKQt_CppSamples/CppSampleManager.cpp#L66-L77

## Type of change

- [x] Bug fix

**Platforms tested on**:

- [ ] Windows
- [ ] Android
- [x] Linux
- [ ] macOS
- [ ] iOS

### CODE REVIEW

File: SampleViewer/SampleManager.cpp [hunk 1]
```diff
@@ -148,23 +148,30 @@ void SampleManager::buildCategoriesList()
...
   {
     qWarning() << "could not find Categories.xml";
   }
+
+#ifdef LOCALSERVER_SUPPORTED
+  if (LocalServer::isInstallValid())
+  {
+    appendCategoryToManager(createCategory("LocalServer", "Local Server", dir));
+  }
+#endif // LOCALSERVER_SUPPORTED
```
Comment (JamesMBallard) [inline SampleViewer/SampleManager.cpp:174]: This block is the fix. The rest are unrelated cleanup.

**Review by jared-2016 — APPROVED** [review]
